ci-operator/step-registry/cucushift/upgrade/toimage/cucushift-upgrade-toimage-commands.sh

```diff
@@ -41,6 +41,71 @@ EOF
     fi
 }
 
+function extract_ccoctl(){
+    echo -e "Extracting ccoctl\n"
+    local retry=5 
+    tmp_ccoctl="/tmp/upgtool"
+    mkdir -p ${tmp_ccoctl}
+    cco_image=$(oc adm release info --image-for='cloud-credential-operator' ${TARGET} -a "${CLUSTER_PROFILE_DIR}/pull-secret") || return 1
+    while ! (env "NO_PROXY=*" "no_proxy=*" oc image extract $cco_image --path="/usr/bin/ccoctl:${tmp_ccoctl}" -a "${CLUSTER_PROFILE_DIR}/pull-secret");
+    do
+        echo >&2 "Failed to extract ccoctl binary, retry..."
+        (( retry -= 1 ))
+        if (( retry < 0 )); then return 1; fi
+        sleep 60
+    done
+    mv ${tmp_ccoctl}/ccoctl /tmp -f
+    if [[ ! -e /tmp/ccoctl ]]; then
+        echo "No ccoctl tool found!" && return 1
+    else
+        chmod 775 /tmp/ccoctl
+    fi
+}
+
+function update_cloud_credentials_oidc(){
+    platform=$(oc get infrastructure cluster -o jsonpath='{.status.platformStatus.type}')
+    preCredsDir="/tmp/pre-include-creds"
+    tobeCredsDir="/tmp/tobe-include-creds"
+    mkdir "${preCredsDir}" "${tobeCredsDir}"
+    # Extract all CRs from live cluster with --included
+    if ! oc adm release extract --to "${preCredsDir}" --included --credentials-requests; then
+        echo "Failed to extract CRs from live cluster!" && exit 1
+    fi
+    if ! oc adm release extract --to "${tobeCredsDir}" --included --credentials-requests "${TARGET}"; then
+        echo "Failed to extract CRs from tobe upgrade release payload!" && exit 1
+    fi
+
+    # TODO: add gcp and azure
+    # Update iam role with ccoctl based on tobeCredsDir
+    if ! diff -r "${preCredsDir}" "${tobeCredsDir}" &> /dev/null; then
+        toManifests="/tmp/to-manifests"
+        mkdir "${toManifests}"
+        case "${platform}" in
+        "AWS")
+            if [[ ! -e ${SHARED_DIR}/aws_oidc_provider_arn ]]; then
+                echo "No aws_oidc_provider_arn file in SHARED_DIR" && exit 1
+            else
+                export AWS_SHARED_CREDENTIALS_FILE="${CLUSTER_PROFILE_DIR}/.awscred"
+                infra_name=${NAMESPACE}-${UNIQUE_HASH}
+                oidc_provider=$(head -n1 ${SHARED_DIR}/aws_oidc_provider_arn)
+                export PATH=/tmp:${PATH}
```

Comment:
I guess we can move this line into `extract_ccoctl` function?

Reply:
I think so, let me updated, and do u think it's worth a re-test since the latest test result passed in https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/43227/rehearse-43227-periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-4.14-upgrade-from-stable-4.13-aws-ipi-disconnected-sts-basecap-none-addcap-f28/1711214278461100032

Reply:
I think we can directly update, no need to re-run the testing, but I suggest to run some other common upgrade jobs to ensure the big change would not break other things.

Reply:
ack, let me restore pre-upgrade and post-upgrade steps and trigger one common upgrade